crates/rules.bzl

```diff
@@ -113,21 +113,31 @@ CrateSummary = provider(
     },
 )
 
+def _is_universe_crate(target):
+    return str(target.label).startswith("@crates__")
+
+def _universe_crate_name(target):
+    return str(target.label).split(".")[0].rsplit("-", 1)[0].removeprefix("@crates__")
```

Comment:
We take everything up to + excluding the first `.`, then take everything up to + excluding the last `-`, then strip `@crates__` from the front.

Reply:
I've taken a look at the Semantic Versioning specification. This should comply with any crate name where the name is proceeded by a valid semantic version.